core/trino-main/src/main/java/io/trino/operator/OutputSpoolingController.java

```diff
@@ -46,12 +49,24 @@ public enum Mode
 
     private Mode currentMode;
 
-    public OutputSpoolingController(boolean inlineInitialRows, long maximumInlinedPositions, long maximumInlinedSize, long initialSpooledSegmentTarget, long maximumSpooledSegmentTarget)
+    private final AtomicLong totalInlinedPositions;
+    private final AtomicLong totalInlinedSize;
```

Comment:
what are thread safety guarantees for this class. 
Something is finishy that some counters are Atomic while other are just longs